### PR TITLE
ARXIVNG-379 Order, page size, query propagate between pages

### PR DESCRIPTION
**Pull request by erickpeirson**
ARXIVNG-379
- After performing a simple search, when the user clicks on the "advanced search" link their query is used to auto-populate the advanced search form.
- When the user selects a page size and/or sort order, that selection is preserved across searches and search routes.

ARXIVNG-380
- Increased all page size options by 100%
- Set default page size to 50

ARXIVNG-382
- Added ``MultiFormatDateField`` to support variable-precision dates.
- Updated placeholder text in date range fields to ``YYYY[-MM[-DD]]``.

image

### CODE REVIEW

**Comment by mhl10** [comment]
When using the variable-precision date fields in advanced search, the lower-bound date is being used for both "From" and "to" when month/day are omitted. E.g. entering "From 2016 to 2017" evaluates to 2016-01-01 to 2017-01-01.  This is OK, though we should probably be clear about the behavior in the tips section, as some users might assume the end date to be upper-bounded. We could do that too, but would require a little more logic for the last day of the month (for February, mainly).

File: lintstats.sh [hunk 1]
```diff
@@ -14,8 +14,7 @@ curl -u $USERNAME:$GITHUB_TOKEN \
 
 
 # Check mypy integration
-pipenv run mypy --ignore-missing-imports -p search | grep -v "test"
-MYPY_STATUS=$?
+MYPY_STATUS=$( pipenv run mypy -p search | grep -v "test.*" | grep -v "defined here" | wc -l | tr -d '[:space:]' )
```
Comment (JaimieMurdock) [inline lintstats.sh]: Much clearer

**Comment by erickpeirson** [comment]
@mhl10 How about something like this:

image

**Comment by mhl10** [comment]
@erickpeirson looks good, probably ok to omit the time and tz.

**Comment by erickpeirson** [comment]
Done. Ok to merge?

File: search/services/index/results.py [hunk 2]
```diff
@@ -165,11 +165,13 @@ def _add_highlighting(result: dict, raw: Response) -> dict:
...
     """Transform an ES search result back into a :class:`.Document`."""
     # typing: ignore
-    result = {'preview': {}}
+    result: Dict[str, Any] = {'preview': {}}
     for key in Document.fields():
         if not hasattr(raw, key):
             continue
         value = getattr(raw, key)
+        if key == 'announced_date_first' and value and isinstance(value, str):
+            value = datetime.strptime(value, '%Y-%m').date()
```
Comment (mhl10) [inline search/services/index/results.py]: Thanks, looks good!

**Pull request merged**